shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs

```diff
@@ -112,33 +114,39 @@ poolRetireInEpoch = withTests (fromIntegral numberOfTests) . property $ do
 
 -- | Check that a `RegPool` certificate properly adds a stake pool.
 registeredPoolIsAdded
-  :: MonadTest m
+  :: forall m
+   . MonadTest m
   => Environment LEDGER
   -> [SourceSignalTarget POOL]
   -> m ()
 registeredPoolIsAdded env ssts =
-  assertAll addedRegPool ssts
+  mapM_ addedRegPool ssts
 
  where
 
   addedRegPool :: SourceSignalTarget POOL
-               -> Bool
+               -> m ()
   addedRegPool sst =
     case signal sst of
       RegPool poolParams -> check poolParams
-      _ -> True
+      _ -> pure ()
    where
-    check :: PoolParams -> Bool
-    check poolParams =
+    check :: PoolParams -> m ()
+    check poolParams = do
       let hk = poolParams ^. poolPubKey
-          pSt = target sst
-          -- PoolParams are registered in pParams map
-       in M.lookup hk (pSt ^. pParams) == Just poolParams
-          -- Hashkey is registered in stPools map
-       && M.lookup hk (pSt ^. stPools . to (\(StakePools x) -> x))
-            == Just (ledgerSlot env)
-          -- Hashkey is registered in cCounters map
-       && hk ∈ M.keys (pSt ^. cCounters)
+          sSt = source sst
+          tSt = target sst
+      -- Check for pool re-registration
+      when (hk ∈ dom (sSt ^. stPools . to (\(StakePools x) -> x))) $ do
+        assert (hk ∈ dom (sSt ^. retiring))
+        assert (hk ∉ dom (tSt ^. retiring))
```

Comment:
I split these changes out into 3 commits so it's easier to see that the only changes here are really the addition of these 4 lines. The rest is simply a transformation of the `Bool` conjunction into a series of monadic assertions.